### PR TITLE
Add migration to `replace-typography-declarations`

### PR DESCRIPTION
**Pull request by aaronccasanova**
Part of #7214

This PR introduces a migration to `replace-typography-declarations` with CSS custom properties.

Running `npx @shopify/polaris-migrator replace-typography-declarations '**/*.scss'` will attempt to update the following declarations:
- `font-family`
- `font-size`
- `font-weight`
- `line-height`

### CODE REVIEW

File: .changeset/stupid-suns-clap.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris-migrator': patch
```
Comment (aaronccasanova) [inline .changeset/stupid-suns-clap.md]:
```suggestion
'@shopify/polaris-migrator': minor
```

**Comment by aaronccasanova** [comment]
Moved back to draft as I am adding more test cases and re-architecting the handlers.

File: polaris-migrator/src/migrations/replace-typography-declarations/tests/replace-typography-declarations.test.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import {check} from '../../../utilities/testUtils';
+
+const migration = 'replace-typography-declarations';
+const fixtures = ['replace-typography-declarations', 'with-namespace'];
+// const fixtures = ['replace-typography-declarations'];
```
Comment (aaronccasanova) [inline polaris-migrator/src/migrations/replace-typography-declarations/tests/replace-typography-declarations.test.ts]:
```suggestion
```

**Review by lgriffee — APPROVED** [review]
Love how clean this code is now based on how you re-architected it! 🧹✨❤️🎉 Going to rearchitect the `replace-sass-lengths` script and new `replace-sass-border` scripts to follow this structure. 👍🏻

File: polaris-migrator/src/migrations/replace-typography-declarations/replace-typography-declarations.ts [hunk 3]
```diff
@@ -0,0 +1,461 @@
...
+  button: {
+    base: '0.9375rem', // 15px
+    'large-screen': '--p-font-size-100',
+  },
+  'button-large': {
+    base: '1.0625rem', // 17px
+    'large-screen': '--p-font-size-200',
+  },
+  'display-x-large': {
+    base: '1.6875rem',
```
Comment (aaronccasanova) [inline polaris-migrator/src/migrations/replace-typography-declarations/replace-typography-declarations.ts]:
```suggestion
    base: '1.6875rem', // 27px
```

**Pull request merged**